icu4c/source/i18n/decimfmt.cpp

```diff
@@ -1673,7 +1685,11 @@ const numparse::impl::NumberParserImpl* DecimalFormat::getParser(UErrorCode& sta
     }
 
     // First try to get the pre-computed parser
+#if U_HAVE_ATOMICS
     auto* ptr = fields->atomicParser.load();
+#else
+    auto* ptr = fields->atomicParser;
```

Comment:
Please just implement the load function in your shim.

Reply:
Do you mean we should add something like `icu::atomic<T>` to shim `std::atomic<T>`?

Reply:
Style-wise, it seems cleaner if your new cfg would just polyfill `std::atomic` with something that works in WebAssembly. I don't think it's useful for code now and into the future to have to figure out how and where to put your atomic ifdefs.

Reply:
I agree. I think you should just implement atomics in a shim instead of trying to disable it everywhere.